converters.go

```diff
@@ -78,18 +79,30 @@ func ParseStringToFloat64(s string, conFailStat *sync.Map) float64 {
 func ParseStringToDecimal(s string, conFailStat *sync.Map) decimal.Decimal {
 	number, err := decimal.NewFromString(s)
 	if err != nil {
-		storeFailure("'"+s+"' asDecimal", conFailStat)
+		StoreFailure("'"+s+"' asDecimal", conFailStat)
 		return decimal.NewFromInt(0)
 	}
 	return number
 }
 
-func ParseStringToInt64(s string, conFailStat *sync.Map) int64 {
+// CheckForError checks the given error and stores a possible failure
+func CheckForError(err error, rawValue interface{}, t reflect.Kind, failStat *sync.Map, fields ...string) {
+	if err != nil {
+		var field string
+		if len(fields) > 0 {
+			field = fmt.Sprintf(`Field %s: `, fields[0])
+		}
+		StoreFailure(fmt.Sprintf(`%sFailed to parse "%s" as %s: %s`, field, rawValue, t.String(), err.Error()), failStat)
+	}
+}
+
+// ParseStringToInt64 parses a string to an int64 and stores any failure
+func ParseStringToInt64(s string, failStat *sync.Map, fields ...string) int64 {
 	number, err := strconv.ParseInt(strings.TrimSpace(s), 10, 64)
 	if err != nil {
 		decimalNumber, err := decimal.NewFromString(s)
 		if err != nil {
-			storeFailure("'"+s+"' asInt64", conFailStat)
+			CheckForError(err, decimalNumber, reflect.Int64, failStat, fields...)
```

Comment:
here should be the string s and not decimalNumer, shouldn't it?
```suggestion
			CheckForError(err, s, reflect.Int64, failStat, fields...)
```